From b3465a4c88258db55e84a981a51d6accc4be5044 Mon Sep 17 00:00:00 2001
From: Jean Christophe Beyler <jean.christophe.beyler@intel.com>
Date: Mon, 3 Jun 2013 15:46:35 -0700
Subject: Dalvik: Klokwork Issue with the backend option

BZ: 113224

Fixing a klokwork issue with the -Xjitbackendoption.

Category: device-enablement
Domain: AOSP-Dalvik-Runtime
Origin: internal
Upstream-Candidate: no, needs rework

Change-Id: If1ef586bfb93daa53f2415667021147453e28026
Orig-MCG-Change-Id: I201d09193bea9cc0c78c59a64d8f60261cb6e422
Signed-off-by: Jean Christophe Beyler <jean.christophe.beyler@intel.com>
Signed-off-by: Qiming Shi <qiming.shi@intel.com>
Signed-off-by: Serguei Katkov <serguei.i.katkov@intel.com>
---
 vm/Init.cpp |   30 +++++++++++-------------------
 1 files changed, 11 insertions(+), 19 deletions(-)

diff --git a/vm/Init.cpp b/vm/Init.cpp
index 416d450..77a7215 100644
--- a/vm/Init.cpp
+++ b/vm/Init.cpp
@@ -1220,6 +1220,7 @@ static int processOptions(int argc, const char* const argv[],
             {
                 //Get =
                 char *equal = strchr (start, '=');
+                char *next = 0;
 
                 if (equal != 0)
                 {
@@ -1228,29 +1229,20 @@ static int processOptions(int argc, const char* const argv[],
 
                     //Next character
                     equal++;
-                }
-                else
-                {
-                    //We strdup so the backend can change the values as it sees fit
-                    equal = 0;
-                }
 
-                //Find end of equal
-                char *next = strchr (equal, ',');
+                    //Find end of equal
+                    next = strchr (equal, ',');
 
-                //If we have another option, we have to separate the current option couple and the next
-                if (next != 0)
-                {
-                    //Set to '\0', separating this key and next key/value couple
-                    *next = '\0';
+                    //If we have another option, we have to separate the current option couple and the next
+                    if (next != 0)
+                    {
+                        //Set to '\0', separating this key and next key/value couple
+                        *next = '\0';
 
-                    //Next character
-                    next++;
-                }
+                        //Next character
+                        next++;
+                    }
 
-                //Set in map: depends if equal is nil or not
-                if (equal != 0)
-                {
                     gDvmJit.backendOptions[start] = equal;
                 }
                 else
-- 
1.7.4.1

